### PR TITLE
Validate parsed courses

### PR DESCRIPTION
**Pull request by mirandayu131**
Find the unique course ID of every course parsed from the other Approved Liberal Studies course list since some courses may be cross-listed. For example, both CS 2110 and ENGRD 2110 would have the same course ID. Courses not found in the course roster data are excluded.

Test plan:
- Check that courses excluded are not in the full list of courses

### CODE REVIEW

File: scripts/other-yes-list/parse.ts
Comment (rohanmaheshwari430) [inline scripts/other-yes-list/parse.ts:104]: could there ever be more than one `course` such that `courseWithId.subject === course.subject && courseWithId.catalogNbr === course.catalogNbr.toString()` is true? because by breaking after you find a match, you are saying that you are only looking for the first occurrence where that `if` condition passes.

**Review by rohanmaheshwari430 — COMMENTED** [review]
nice work miranda!

File: scripts/other-yes-list/parse.ts
Comment (rohanmaheshwari430) [inline scripts/other-yes-list/parse.ts:95]: could there ever be more than one `course` such that `courseWithId.subject === course.subject && courseWithId.catalogNbr === course.catalogNbr.toString()` is true? because by breaking after you find a match, you are saying that you are only looking for the first occurrence where that `if` condition passes.

**Review by zachary0kent — CHANGES_REQUESTED** [review]
Great work! Can you change the base branch to `master` and wait until `parse-lib-studies` is merged to merge this?

File: scripts/other-yes-list/parse.ts
Comment (zachary0kent) [inline scripts/other-yes-list/parse.ts:107]:
```suggestion
  for (const { subject, catalogNbr, crseId } of approvedCourses) {
    for (const course of courseEntries) {
      if (subject === course.subject && catalogNbr === course.catalogNbr.toString()) {
        validCourses.push({ ...course, crseId });
        break;
      }
    }
  }
```

**Review by zachary0kent — CHANGES_REQUESTED** [review]
Can you merge master into your branch? Then I think this should be good to merge :)

**Pull request closed without merging**
